src/Rector/Class_/CommandPropertyToAttributeRector.php

```diff
@@ -118,34 +203,74 @@ public function refactor(Node $node): ?Node
             }
         }
 
-        $node->attrGroups[] = $this->createAttributeGroup($defaultName, $defaultDescription);
-
-        return $node;
+        return $defaultDescription;
     }
 
-    private function createAttributeGroup(string $defaultName, ?string $defaultDescription): AttributeGroup
+    private function resolveAliases(Class_ $class): ?Array_
     {
-        $attributeGroup = $this->phpAttributeGroupFactory->createFromClass(self::ATTRIBUTE);
+        $commandAliases = null;
+        $this->traverseNodesWithCallable($class->stmts, function (Node $node) use (&$commandAliases) {
+            if (! $node instanceof MethodCall) {
+                return null;
+            }
 
-        $attributeGroup->attrs[0]->args[] = new Arg(new String_($defaultName));
+            if (! $this->isObjectType($node->var, new ObjectType('Symfony\Component\Console\Command\Command'))) {
+                return null;
+            }
 
-        if ($defaultDescription !== null) {
-            $attributeGroup->attrs[0]->args[] = new Arg(new String_($defaultDescription));
-        }
+            if (! $this->isName($node->name, 'setAliases')) {
+                return null;
+            }
 
-        return $attributeGroup;
+            if (! $node->getArgs()[0]->value instanceof Array_) {
+                return null;
+            }
+
+            $commandAliases = $node->getArgs()[0]
+->value;
+
+            $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
+            if ($parentNode instanceof MethodCall) {
+                $parentNode->var = $node->var;
+            } else {
+                $this->removeNode($node);
+            }
+        });
+
+        return $commandAliases;
     }
 
-    private function getValueFromProperty(Property $property): ?string
+    private function resolveHidden(Class_ $class): ?ConstFetch
     {
-        if (count($property->props) !== 1) {
-            return null;
-        }
-        $propertyProperty = $property->props[0];
-        if ($propertyProperty->default instanceof String_) {
-            return $propertyProperty->default->value;
-        }
+        $commandHidden = null;
+        $this->traverseNodesWithCallable($class->stmts, function (Node $node) use (&$commandHidden) {
+            if (! $node instanceof MethodCall) {
+                return null;
+            }
 
-        return null;
+            if (! $this->isObjectType($node->var, new ObjectType('Symfony\Component\Console\Command\Command'))) {
+                return null;
+            }
+
+            if (! $this->isName($node->name, 'setHidden')) {
+                return null;
+            }
+
+            if (! $node->getArgs()[0]->value instanceof ConstFetch) {
+                return null;
+            }
+
+            $commandHidden = $node->getArgs()[0]
```

Comment:
multiple `getArgs()` can be saved to variable, also, if you have `$node->isFirstCallable()` check, you can just use `$node->args`, the `@var Arg $firstArg` is fine after that

Reply:
I defined a variable for getArgs, i had a look at `$node->isFirstCallable()` but its not a function on `$node`?

Reply:
Ah its probably `isFirstClassCallable`

Reply:
Was discussed in https://github.com/rectorphp/rector-symfony/pull/279/files#r1029199016

Reply:
Its ready now